frontend/javascript/components/postupload/post-upload.vue

```diff
@@ -0,0 +1,1494 @@
+  props.form.fields.publicbody?.initial?.id?.toString() ||
+  props.object_public_body_id
+
+const formPublicbodyLabel = props.object_public_body?.name || 'Error'
+
+const documentsSelectedPdfRedaction = ref([])
+const documentsPdfRedactionIndex = computed(() => {
+  if (3000 < step.value && step.value < 3100) return step.value - 3001
+  return false
+})
+
+const currentPdfRedactionDoc = computed(() => {
+  if (documentsPdfRedactionIndex.value === false) return
+  return documentsSelectedPdfRedaction.value[documentsPdfRedactionIndex.value]
+})
+
+const documentUploader = ref()
+const documentsImagesConverting = ref(false)
+const documentsConvertImages = () => {
+  if (documentUploader.value.$refs.imageDocument.length === 0) {
+    console.log('no images to convert')
+    gotoStep()
+    return
+  }
+  documentsImagesConverting.value = true
+  // XXX wild mix of Vue2&3 refs, not the cleanest, but a bus was basically impossible
+  // we are calling a grand-child component's method
+  documentUploader.value.$refs.imageDocument?.[0]?.convertImages?.()
+  // ^child              ^vue2 ^grandchild...REN!  ^method
+  // alternative: pass a prop, watch it, react?
+}
+const documentsImagesDocumentFilenameDefault = 'brief.pdf'
+const documentsImagesDocumentFilename = ref(
+  documentsImagesDocumentFilenameDefault
+)
+const documentsImagesDocumentFilenameNormalized = computed(() =>
+  documentsImagesDocumentFilename.value.replace(/\.pdf$/, '')
+)
+
+const documentsBasicOperations = ref(false)
+
+const pdfRedaction = ref()
+const pdfRedactionCurrentHasRedactions = ref(false)
+const pdfRedactionProcessing = ref(false)
+const pdfRedactionRedact = () => {
+  pdfRedactionProcessing.value = true
+  // XXX calling child's method
+  // alternatively, could listen to an event
+  pdfRedaction.value
+    .redactOrApprove()
+    // .then(delay(3000))
+    .then(() => {
+      pdfRedactionProcessing.value = false
+      pdfRedactionCurrentHasRedactions.value = false
+      gotoStep()
+    })
+}
+const pdfRedactionUploaded = () => {
+  // TODO handle errors?
+  // XXX again, we're calling child's method, could handle attachments here, store-like,
+  //   and pass as a "override prop" to the components
+  documentUploader.value.refreshAttachments()
+}
+
+const documentUploaderSelectAll = (val) => {
+  documentUploader.value.setAllSelect(val)
+}
+
+const scrollHeaderIntoViewIfNecessary = () => {
+  // we could bail early if !isDesktop.value,
+  // but this check fill automagically still work as the header is always in viewport on mobile
+  // alt: check appshell-header-container--desktop, if hidden all rect attributes == 0
+  const header = document.querySelector('.appshell-header')
+  const rect = header.getBoundingClientRect()
+  if (rect.bottom < 0) {
+    // timeout to make it feel less jarring
+    setTimeout(() => header.scrollIntoView({ behavior: 'smooth' }), 500)
+  }
+}
+
+const mobileHeaderTitle2 = computed(() => {
+  switch (step.value) {
+    case 1100:
```

Comment:
where do the numbers come from?

Reply:
For the WIP I've used the autogenerated frame IDs/numbers from Figma, ähem. Replacing them with something better would go hand-in-hand with a proper state machine :) (but will require some reworking of the step-ID-to-file-index arithmetic...)